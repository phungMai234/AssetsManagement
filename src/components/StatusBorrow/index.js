import React, { useMemo } from 'react';
import styled from 'styled-components';
import { BORROWING, BORROWING_TEXT, CLOSED, CLOSED_TEXT, DRAFT, DRAFT_TEXT } from 'utils/constant';

const Wrapper = styled.div`
  .draft {
    color: #ffc107;
  }

  .borrowing {
    color: #28a745;
  }

  .canceled {
    color: #dc3545;
  }

  .closed {
    color: #6c757d;
  }
`;

const StatusBorrow = ({ status }) => {
  const textStatus = useMemo(() => {
    switch (status) {
      case BORROWING:
        return BORROWING_TEXT;
      case CLOSED:
        return CLOSED_TEXT;
      case DRAFT:
        return DRAFT_TEXT;
      default:
        return '';
    }
  }, [status]);

  return (
    <Wrapper>
      <div className={status}>{textStatus}</div>
    </Wrapper>
  );
};

export default StatusBorrow;
