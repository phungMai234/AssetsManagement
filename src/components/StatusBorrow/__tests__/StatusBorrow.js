import React from 'react';

import { render } from '@testing-library/react';

import StatusBorrow from '../StatusBorrow';

describe('StatusBorrow', () => {
  it('should render correctly', () => {
    const container = render(<StatusBorrow />);
    expect(container.container).toMatchSnapshot();
  });

  it('should render correctly with status draft', () => {
    const container = render(<StatusBorrow status="draft" />);
    expect(container.container).toMatchSnapshot();
  });

  it('should render correctly with status borrowing', () => {
    const container = render(<StatusBorrow status="borrowing" />);
    expect(container.container).toMatchSnapshot();
  });

  it('should render correctly with status closed', () => {
    const container = render(<StatusBorrow status="closed" />);
    expect(container.container).toMatchSnapshot();
  });
});
