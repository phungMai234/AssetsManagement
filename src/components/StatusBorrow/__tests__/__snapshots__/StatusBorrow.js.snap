// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBorrow should render correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ cAtJaJ"
  >
    <div />
  </div>
</div>
`;

exports[`StatusBorrow should render correctly with status borrowing 1`] = `
<div>
  <div
    class="sc-bdfBwQ cAtJaJ"
  >
    <div
      class="borrowing"
    >
      Đang mượn
    </div>
  </div>
</div>
`;

exports[`StatusBorrow should render correctly with status closed 1`] = `
<div>
  <div
    class="sc-bdfBwQ cAtJaJ"
  >
    <div
      class="closed"
    >
      Đã trả
    </div>
  </div>
</div>
`;

exports[`StatusBorrow should render correctly with status draft 1`] = `
<div>
  <div
    class="sc-bdfBwQ cAtJaJ"
  >
    <div
      class="draft"
    >
      Bản nháp
    </div>
  </div>
</div>
`;
